rust_src/src/lisp.rs

```diff
@@ -106,11 +142,38 @@ impl<T> ExternalPtr<T> {
     pub const fn new(p: *mut T) -> Self {
         Self(p)
     }
+
+    pub fn is_null(self) -> bool {
+        self.0.is_null()
+    }
+
+    pub const fn as_ptr(self) -> *const T {
+        self.0
+    }
+
+    pub fn as_mut(&mut self) -> *mut T {
+        self.0
+    }
+
+    pub fn from_ptr(ptr: *mut c_void) -> Option<Self> {
+        unsafe { ptr.as_ref().map(|p| mem::transmute(p)) }
+    }
 }
 
-pub type LispSubrRef = ExternalPtr<Aligned_Lisp_Subr>;
-impl LispSubrRef {
-    pub fn as_mut(self) -> *mut Aligned_Lisp_Subr {
-	self.0
+impl<T> Deref for ExternalPtr<T> {
+    type Target = T;
+    fn deref(&self) -> &Self::Target {
+        unsafe { &*self.0 }
+    }
+}
+
+impl<T> DerefMut for ExternalPtr<T> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        unsafe { &mut *self.0 }
     }
 }
+
+// Lisp_Subr support
+
+pub type LispSubrRef = ExternalPtr<Aligned_Lisp_Subr>;
+unsafe impl Sync for Aligned_Lisp_Subr {}
```

Comment:
This was in the old code base ?

Reply:
There was an impl Sync for Lisp_Subr in the old codebase, however that lived in remacs_sys.rs. I did add equivalent code for Aligned_Lisp_Subr implementing Sync in remacs_sys, so I don't think we need that here. We should still have `pub type LispSubrRef = ExternalPtr<Aligned_Lisp_Subr>;` here however

Reply:
> I did add equivalent code for Aligned_Lisp_Subr implementing Sync in remacs_sys, so I don't think we need that here

Ah I missed that.